### PR TITLE
Adding an interactivity tutorial.

### PR DESCRIPTION
**Pull request by Sfshaza**
This tutorial is not complete, but I want the PR up in time for the Flutter Hackathon tomorrow.

If the code is approved soon ( https://github.com/flutter/website/pull/408 ), I will update the links in the tutorial.

Feedback welcome!

@sethladd 

https://sz-flutter.firebaseapp.com/tutorials/layout/

### CODE REVIEW

**Comment by sethladd** [comment]
Re-kicked the Travis build.

File: tutorials/interactive/index.md [hunk 1]
```diff
@@ -0,0 +1,781 @@
...
+  * tapbox-a
+  * tapbox-b
+  * tapbox-c
+{% endcomment %}
+
+## Introduction
+
+How do you modify your app to make it react to user input? In
+[Building Layouts for Flutter](https://flutter.io/tutorials/layout/),
+we showed how to create the layout for the Lakes app:
```
Comment (sethladd) [inline tutorials/interactive/index.md]: "the Lakes app" makes it sound official. Not sure if the reader will know what _the_ Lakes app is.

Maybe, "for the following screenshot:"

File: tutorials/interactive/index.md [hunk 2]
```diff
@@ -0,0 +1,781 @@
...
+
+<center><img src="images/lakes.png" style="border:1px solid black" alt="The starting Lakes app that we will modify"></center><br>
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Let's use CSS to center and create space after the image. I think bootstrap even has a class to center something.

File: tutorials/interactive/index.md [hunk 3]
```diff
@@ -0,0 +1,781 @@
...
+
+In this tutorial, you'll add interactivity to the Lakes app.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Is this a better way to start this article? Stating what this doc does and what you'll learn.

Then, we talk about the context (here's the screenshot, here's the star, etc)

(and s/the Lakes app/a simple screen for a mobile app)

(then, later, you can show the screenshot and connect it back to the other tutorial)

File: tutorials/interactive/index.md [hunk 4]
```diff
@@ -0,0 +1,781 @@
...
+TODO:
+
+* Restore _private variables in tapbox examples.
+* Add the following examples to the GitHub repo, and update all links
+  * lakes-interactive
+  * tapbox-a
+  * tapbox-b
+  * tapbox-c
+{% endcomment %}
+
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Perhaps for a "what you'll learn"

* Respond to taps
* Create a custom widget
* The difference between a stateless and stateful widget

File: tutorials/interactive/index.md [hunk 5]
```diff
@@ -0,0 +1,781 @@
...
+* [`lib/main.dart`](https://raw.githubusercontent.com/flutter/website/master/_includes/_code/lakes/main.dart)
+* [`pubspec.yaml`](https://raw.githubusercontent.com/flutter/website/master/_includes/_code/lakes/pubspec.yaml)
+* [`lakes.jpg`](https://github.com/flutter/website/blob/master/_includes/_code/lakes/images/lake.jpg)
+
+<aside class="alert alert-success" markdown="1">
+<i class="fa fa-lightbulb-o"> </i> **Tip:**
+If you are new to Flutter, use these steps to prepare for this tutorial:
+
+* [Set up](https://flutter.io/setup/) your environment.
+* If you don't have a preferred IDE, we recommend
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Not sure we need to say this? It's covered in Set Up (I think)

File: tutorials/interactive/index.md [hunk 6]
```diff
@@ -0,0 +1,781 @@
...
+  [IntelliJ community edition.](https://flutter.io/intellij-setup/) (It's free!)
+* Save the files in the list above to your computer.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Can you link here (instead of saying: above)

File: tutorials/interactive/index.md [hunk 7]
```diff
@@ -0,0 +1,781 @@
...
+* [Create your first Flutter
+  app.](https://flutter.io/getting-started/#creating-your-first-flutter-app)
+* Replace the `lib/main.dart` and `pubspec.yaml` files in your project
```
Comment (sethladd) [inline tutorials/interactive/index.md]: link these files?

File: tutorials/interactive/index.md [hunk 8]
```diff
@@ -0,0 +1,781 @@
...
+  with those provided.
+* Create an `images` directory in your project, and add `lakes.jpg`.
+* Once you have a connected and enabled device, or you've launched the
```
Comment (sethladd) [inline tutorials/interactive/index.md]: s/and/an ?

File: tutorials/interactive/index.md [hunk 9]
```diff
@@ -0,0 +1,781 @@
...
+  * [Standard widgets](#standard-widgets)
+  * [Material widgets](#material-widgets)
+* [Resources](#resources)
+
+<a name="stateful-stateless"></a>
+## Stateful and stateless widgets
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Can we use CSS instead of the `<b>`

File: tutorials/interactive/index.md [hunk 10]
```diff
@@ -0,0 +1,781 @@
...
+
+* _Stateful_ means that a widget can change because of user interaction or
+  other factors, such a updating the UI using information from a service.
+* All widgets are either stateful or stateless.
+* A stateless widget sits there and looks pretty.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: hah This made me LOL.

Not sure it's quite accurate... an animation could move a stateless widget around the screen every frame.

File: tutorials/interactive/index.md [hunk 11]
```diff
@@ -0,0 +1,781 @@
...
+* A State object manages the stateful widget's state.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Not quite sure if "manages" is the right word. Maybe, "contains"?

File: tutorials/interactive/index.md [hunk 12]
```diff
@@ -0,0 +1,781 @@
...
+{% comment %}
+* You don't need to worry about abandoned widgets&mdash;they are automatically
```
Comment (sethladd) [inline tutorials/interactive/index.md]: what does "abandoned" mean? Did we define this term previously?

File: tutorials/interactive/index.md [hunk 13]
```diff
@@ -0,0 +1,781 @@
...
+  collected by the garbage collector (GC).
+{% endcomment %}
+* All widgets are _immutable_, meaning once created they cannot change.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Maybe: "should be immutable". Right now, we don't have a way to enforce immutability. We hope to get it soon. But this is a design goal, not a fundamental rule.

File: tutorials/interactive/index.md [hunk 14]
```diff
@@ -0,0 +1,781 @@
...
+* When a widget's appearance changes, a new widget is created to reflect the change,
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Does this sentence apply only to stateful widgets? Should we clarify that?

File: tutorials/interactive/index.md [hunk 15]
```diff
@@ -0,0 +1,781 @@
...
+  but the State object allows the widget's state to persist.
+* Typically, the State object is managed by the widget itself, the parent widget,
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Should we define "managed" ? As far as I know (and, I admit, it's not that far), the StatefulWidget creates the State object. But I don't know what else it does to "manage" it

File: tutorials/interactive/index.md [hunk 16]
```diff
@@ -0,0 +1,781 @@
...
+  or some combination.
+* The widget's implementation determines who manages the state.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: The implementation by me (the widget author), or the implementation of the widget from the framework authors? This wasn't clear.

File: tutorials/interactive/index.md [hunk 17]
```diff
@@ -0,0 +1,781 @@
...
+
+</div>
+
+There are two kinds of widgets, stateful and stateless.
+
+A _stateless_ widget
+is drawn to the screen and sits there, informing the user,
```
Comment (sethladd) [inline tutorials/interactive/index.md]: what does "sits there" mean?

File: tutorials/interactive/index.md [hunk 18]
```diff
@@ -0,0 +1,781 @@
...
+or making the UI look more attractive.
+The user doesn't interact with a stateless widget.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Not sure this is always true. For example, IconButton is a StatelessWidget.

File: tutorials/interactive/index.md [hunk 19]
```diff
@@ -0,0 +1,781 @@
...
+[Icon,](https://docs.flutter.io/flutter/material/Icon-class.html)
+[Image,](https://docs.flutter.io/flutter/widgets/Image-class.html) and
+[Text](https://docs.flutter.io/flutter/widgets/Text-class.html) are
+examples of stateless widgets, which subclass
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html).
+
+A _stateful_ widget is dynamic. The user can interact with it
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Maybe: "can have state that lives beyond the life of the widget itself."

File: tutorials/interactive/index.md [hunk 20]
```diff
@@ -0,0 +1,781 @@
...
+[Slider](https://docs.flutter.io/flutter/material/Slider-class.html) are
+examples of stateful widgets, which subclass
+[StatefulWidget](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html).
+
+<a name="stateful-widget-pattern-1"></a>
+## Creating a stateful widget
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
```
Comment (sethladd) [inline tutorials/interactive/index.md]: If we can use CSS for `<b>`, that would be excellent.

File: tutorials/interactive/index.md [hunk 21]
```diff
@@ -0,0 +1,781 @@
...
+
+* To create a custom stateful widget, subclass StatefulWidget and State.
+* The State object manages the widget's state.
+* Calling `setState()` updates the widget's UI.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: maybe: "tells the framework to redraw the widget"

File: tutorials/interactive/index.md [hunk 22]
```diff
@@ -0,0 +1,781 @@
...
+
+In this section, you'll create a custom stateful widget.
+You'll be replacing two stateless widgets,
+the solid red star and the numeric count next to it, with a single custom
+stateful widget that manages a row with two children widgets: an IconButton
+and Text.
+
+A custom stateful widget requires creating two classes:
+
+1. A subclass of StatefulWidget that defines the widgets and describes how to
```
Comment (sethladd) [inline tutorials/interactive/index.md]: s/widgets/widget

or, what do you mean: "defines the widgets" ?

File: tutorials/interactive/index.md [hunk 23]
```diff
@@ -0,0 +1,781 @@
...
+   draw them in its `build()` method.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: The State class has the build() method. Yes, that's confusing. :( But, the StatefulWidget doesn't have the build() method

File: tutorials/interactive/index.md [hunk 24]
```diff
@@ -0,0 +1,781 @@
...
+2. A subclass of State that manages state for that widget.
+
+This section shows you how to build a stateful widget, called FavoriteWidget,
```
Comment (sethladd) [inline tutorials/interactive/index.md]: when do we use "stateful widget" and when do we use StatefulWidget ?

File: tutorials/interactive/index.md [hunk 25]
```diff
@@ -0,0 +1,781 @@
...
+for the Lakes app. The first step is choosing who manages the State object.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: do you mean, "how state is managed?"

Only the StatefulWidget creates an instance of State.

File: tutorials/interactive/index.md [hunk 26]
```diff
@@ -0,0 +1,781 @@
...
+
+<a name="step-1"></a>
+### Step 1: Decide who manages the widget's state
+
+There are different approaches to managing a widget's state, but in our example
+the widget itself, FavoriteWidget, will manage its own state.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Technically, the State object contains the state. Not sure it's totally accurate to say "FavoriteWidget manages its own state", unless you mean an instance of State.

File: tutorials/interactive/index.md [hunk 27]
```diff
@@ -0,0 +1,781 @@
...
+<a name="step-3"></a>
+### Step 3: Subclass the State object
+
+The custom State class stores the mutable information&mdash;the logic and internal
+state that can change over the lifetime of the widget. When the app first launches,
+the UI displays a solid red star, indicating that the lake has "favorite" status,
+and has a total of 41 “likes”. The State object stores this information in the
+`isFavorited` and `favoriteCount` variables.
+
+The widget’s `build` method creates a packed row containing a red IconButton,
```
Comment (sethladd) [inline tutorials/interactive/index.md]: what is a "packed row" ? Is that a class?

File: tutorials/interactive/index.md [hunk 28]
```diff
@@ -0,0 +1,781 @@
...
+and Text.  The widget uses
+[IconButton,](https://docs.flutter.io/flutter/material/IconButton-class.html)
+instead of Icon, because its `onPressed` property defines the callback method
```
Comment (sethladd) [inline tutorials/interactive/index.md]: what is "its" here? Icon or IconButton?

File: tutorials/interactive/index.md [hunk 29]
```diff
@@ -0,0 +1,781 @@
...
+    );
+
+    return new MaterialApp(
+      // ...
+    );
+  }
+}
+{% endprettify %}
+
+<br>That's it! When you hot reload the app, the star icon should now respond to taps.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: Can we use CSS to create more space after code blocks? (instead of resorting to `<br>`)

File: tutorials/interactive/index.md [hunk 30]
```diff
@@ -0,0 +1,781 @@
...
+
+The rest of this page covers several ways a widget's state can be managed,
+and lists other available interactive widgets.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: But then we go into troubleshooting. :) Maybe add "and troubleshooting tips"

File: tutorials/interactive/index.md [hunk 31]
```diff
@@ -0,0 +1,781 @@
...
+
+### Problems?
+
+If you can't get your code to run, look in your IDE for possible errors.
+[Debugging Flutter Apps](https://flutter.io/debugging/) might help.
+If you still can't find the problem,
+check your code against the interactive Lakes example on GitHub.
```
Comment (sethladd) [inline tutorials/interactive/index.md]: And maybe ask the reader to join our mailing list and ask there?

**Comment by Sfshaza** [comment]
Ready for review, @Hixie and @sethladd!

Staged:  https://sz-flutter.firebaseapp.com/tutorials/interactive/

File: tutorials/interactive/index.md [hunk 32]
```diff
@@ -0,0 +1,814 @@
...
+
+## Introduction
+
+How do you modify your app to make it react to user input?
+In this tutorial, you'll add interactivity to an app contains only
+non-interactive widgets.  Specifically, you'll modify an icon to make
+it tappable.  To accomplish this, you'll create a custom stateful widget
+that manages two stateless widgets.
+
+<a name="note"></a>
```
Comment (Hixie) [inline tutorials/interactive/index.md]: Pretty sure <a name> is obsolete HTML. Just put an id="name" on the `aside`?

File: tutorials/interactive/index.md [hunk 33]
```diff
@@ -0,0 +1,814 @@
...
+  from GitHub.
+
+Once you have a connected and enabled device, or you've launched the
+[iOS simulator](/setup/#set-up-the-ios-simulator) (part of the Flutter install),
+you are good to go!
+</aside>
+
+
+In [Building Layouts for Flutter](https://flutter.io/tutorials/layout/),
+we showed how to create the layout for the followng screenshot.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: typo in following

File: tutorials/interactive/index.md [hunk 34]
```diff
@@ -0,0 +1,814 @@
...
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* All widgets are either stateful or stateless.
+* _State_ means information about a widget that might change.
+  A checkbox's state, for example, would probably
+  include a boolean indicating whether it's currently checked.
+* If a widget changes based on user interaction, or as the result of a
+  data feed (for example), it's _stateful_.
+* All widgets should be _immutable_, meaning once created they cannot change.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: But in the previous bullet point you just said that a widget could change.

I think this is confusing "class that inherits from Widget" with "user interaction control", both of which could be called "widget".

**Comment by sethladd** [comment]
Thanks!

* the side nav scrolls off the screen for your stage. Weird?

* "What’s the point?" for "Stateful and stateless widgets" is pretty long. There's a lot of points in there. Does that mean we need two sections? Or, are there points in here that aren't necessary key?

* "The class managing the state object tells the widget to update the UI by calling setState." -- isn't so much "the class managing the state object", because the StatefulWidget doesn't have setState, and the State object isn't managing the state object, it _is_ the state object :)

* "A stateless widget informs the user, or makes the UI look more attractive" - I think the same thing can be said about a stateful widget.

File: tutorials/interactive/index.md [hunk 35]
```diff
@@ -0,0 +1,814 @@
...
+* Being immutable is fine for a stateless widget, because its appearance
+  never changes.
+* When a stateful widget changes (its slider moved, for example),
+  the widget needs to rebuild in order to update the UI.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: Actually the Slider widget doesn't have to be stateful (it actually is, but only because it can animate its bubble, not because of being interactive). Any time you do anything on a slider, we invoke a callback, which should make _another_ widget (which should be stateful) call setState to update its state, which we then pass back into the Slider when the system rebuilds. The slider doesn't itself keep track of the changed state.

File: tool/LICENSE [hunk 36]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
```
Comment (kwalrath) [inline tool/LICENSE]: 2014 -> 2017?

File: tutorials/interactive/index.md [hunk 37]
```diff
@@ -0,0 +1,814 @@
...
+* A _State_ object holds the changeable information for a stateful widget,
+  separating the widget's state from its appearance.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: I'm not really sure what this means. The appearance of the widget is set by its `build` function, which is in the `State` for a `StatefulWidget` and in the widget for a `StatelessWidget`.

File: tutorials/interactive/index.md [hunk 38]
```diff
@@ -0,0 +1,814 @@
...
+* The class managing the state object tells the widget to update the UI by
+  calling `setState`.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: The class managing the state object is the State object. `setState` should only be called by the State object.

File: tutorials/interactive/index.md [hunk 39]
```diff
@@ -0,0 +1,814 @@
...
+* Typically, the state object is managed by the widget itself, the parent
+  widget, or some combination.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: I don't know what this means.

File: tutorials/interactive/index.md [hunk 40]
```diff
@@ -0,0 +1,814 @@
...
+* How the widget is implemented determines who manages its state.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: Or this. What do you mean by "manage"?

File: tutorials/interactive/index.md [hunk 41]
```diff
@@ -0,0 +1,814 @@
...
+
+</div>
+
+There are two kinds of widgets, stateful and stateless.
```
Comment (Hixie) [inline tutorials/interactive/index.md]: There are at least two other kinds of widgets, inheritedwidget and renderobjectwidget. There could also be others.

File: tutorials/interactive/index.md [hunk 42]
```diff
@@ -0,0 +1,814 @@
...
+
+A _stateless_ widget informs the user, or makes the UI look more attractive.
+A stateless widget has no internal state to manage.
+[Icon,](https://docs.flutter.io/flutter/material/Icon-class.html)
+[Image,](https://docs.flutter.io/flutter/widgets/Image-class.html) and
```
Comment (Hixie) [inline tutorials/interactive/index.md]: Image is stateful.

File: tutorials/interactive/index.md [hunk 43]
```diff
@@ -0,0 +1,814 @@
...
+[Text](https://docs.flutter.io/flutter/widgets/Text-class.html) are
+examples of stateless widgets, which subclass
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html).
+
+A _stateful_ widget is dynamic and can have state that lives beyond the
+life of the widget itself. The user can interact with a stateful widget
```
Comment (Hixie) [inline tutorials/interactive/index.md]: I'm not sure what this means.

StatefulWidget subclasses have no lifetime. They are immutable and therefore conceptually live forever.
A control in the element tree that is represented by a StatefulElement has a single State and a series of StatefulWidgets throughout its lifetime.

File: tutorials/interactive/index.md [hunk 44]
```diff
@@ -0,0 +1,814 @@
...
+(by moving a slider, or checking a box, for example),
```
Comment (Hixie) [inline tutorials/interactive/index.md]: moving a slider and checking a box are both examples of things that result in state changing in State objects for widgets that are unrelated to the controls themselves (except for the animation side of things).

File: tutorials/interactive/index.md [hunk 45]
```diff
@@ -0,0 +1,814 @@
...
+
+How do you modify your app to make it react to user input?
+In this tutorial, you'll add interactivity to an app contains only
+non-interactive widgets.  Specifically, you'll modify an icon to make
+it tappable.  To accomplish this, you'll create a custom stateful widget
+that manages two stateless widgets.
+
+<a name="note"></a>
+<aside class="alert alert-info" markdown="1">
+<i class="fa fa-lightbulb-o"> </i> **Note:**
```
Comment (Hixie) [inline tutorials/interactive/index.md]: The word "Note" doesn't really add anything here IMHO. I would remove prefixes that say "Note" throughout. That's just my opinion though, feel free to disregard. :-)

Comment (kwalrath) [inline tutorials/interactive/index.md]: The nice thing about "Note:" is that you can search for it.

File: tutorials/interactive/index.md [hunk 46]
```diff
@@ -0,0 +1,814 @@
...
+should redraw. The `toggleFavorite` function swaps the UI between
+1) a star icon and the number ‘41’, and
+2) a star_border icon and the number ‘40’.
+
+<!-- _code/lakes-interactive/main.dart -->
+<!-- skip -->
+{% prettify dart %}
+class FavoriteWidgetState extends State<FavoriteWidget> {
+  [[highlight]]bool _isFavorited = true;[[/highlight]]
+  [[highlight]]int favoriteCount = 41;[[/highlight]]
```
Comment (Hixie) [inline tutorials/interactive/index.md]: It's weird that _isFavorited is private (which is good) but favoriteCount is not. I'd make them both private, along with toggleFavorite.

File: tutorials/interactive/index.md [hunk 47]
```diff
@@ -0,0 +1,814 @@
...
+The only file you need is
+[lib/main.dart]().
+[PENDING: Add a link once it's up on the site.]
+</aside>
+{% endcomment %}
+
+How do you decide which approach to use?
+
+The general rule of thumb is that the parent widget should manage the
+state until it becomes too much of a burden. If there is any state
```
Comment (Hixie) [inline tutorials/interactive/index.md]: Here's another rule of thumb which I think may be useful:

- If the state in question is user data, for example the checked/unchecked mode of a checkbox or the position of a slider, then the state is best managed by the parent widget.
- If the state in question is aesthetic, for example an animation, then the state is best managed by the widget itself.

File: tutorials/interactive/index.md [hunk 48]
```diff
@@ -0,0 +1,814 @@
...
+[`lib/main.dart`](https://raw.githubusercontent.com/flutter/website/master/_includes/_code/tapbox-b/main.dart)
+
+<aside class="alert alert-success" markdown="1">
+<i class="fa fa-lightbulb-o"> </i> **Tip:**
+When creating API, consider using the `@required` annotation for
+any parameters that your code relies on.
+To use `@required`, import the meta library:
+
+<pre>
+import package: meta/meta.dart;
```
Comment (Hixie) [inline tutorials/interactive/index.md]: This is missing quotes, and has an extraneous space.

Also, I would recommend `'package:flutter/foundation.dart'` instead, which reexports meta.

**Comment by Hixie** [comment]
Left some comments.

As a general point, is there a reason we're adding this kind of thing to the Web site rather than to the API docs themselves? I would encourage putting this kind of discussion right into the dartdocs, and then just pointing people to them from the web site.

File: tutorials/interactive/index.md (hunk 47 above)
Comment (Sfshaza) [inline tutorials/interactive/index.md]: That's gold, thanks!

**Comment by Sfshaza** [comment]
@sethladd , further CSS tweaking can happen after First Look.

Ian, perhaps we should schedule a discussion of API vs website docs in a Flutter docs sync, with @LarkAscending and @sethladd?

Thanks so much for your quick and thorough review!!!  Please review, @Hixie.

File: tutorials/interactive/index.md [hunk 49]
```diff
@@ -0,0 +1,804 @@
...
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* To create a custom stateful widget, subclass two classes:
+  StatefulWidget and State.
+* The state object contains the widget's state, and the widget's `build()`
+  method.
+* The state object calls `setState()`,
```
Comment (Hixie) [inline tutorials/interactive/index.md]: maybe "occasionally calls"?

File: tutorials/interactive/index.md [hunk 50]
```diff
@@ -0,0 +1,804 @@
...
+The custom State class stores the mutable information&mdash;the logic and
+internal state that can change over the lifetime of the widget.
+When the app first launches, the UI displays a solid red star,
+indicating that the lake has "favorite" status, and has a total of 41 “likes”.
+The state object stores this information in the
+`_isFavorited` and `_favoriteCount` variables.
+
+The state object also defines the `build` method. This `build` method
+creates a row containing a red IconButton, and Text.  The widget uses
+[IconButton,](https://docs.flutter.io/flutter/material/IconButton-class.html)
```
Comment (Hixie) [inline tutorials/interactive/index.md]: nit: here and elsewhere, I would recommend putting the comma after the `[]()` sequence, so that there's no risk that the reader will think the comma is part of the class name.

File: tutorials/interactive/index.md [hunk 51]
```diff
@@ -0,0 +1,804 @@
...
+<hr>
+
+<a name="parent-managed"></a>
+### The parent widget manages the widget's state
+
+Often it makes the most sense for the parent widget to manage the state
+and tell its child widget when to update. For example,
+[IconButton](https://docs.flutter.io/flutter/material/IconButton-class.html)
+allows you to treat an icon as a tappable button.
+IconButton is a stateless widget because the designers decided that
```
Comment (Hixie) [inline tutorials/interactive/index.md]: Maybe replace "the designers" with "we"?

File: tutorials/interactive/index.md [hunk 52]
```diff
@@ -0,0 +1,804 @@
...
+[StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html).
+
+A _stateful_ widget is dynamic. The user can interact with a stateful widget
+(by moving a slider, or checking a box, for example),
+or it changes over time (perhaps a data feed causes the UI to update).
+[Checkbox,](https://docs.flutter.io/flutter/material/Checkbox-class.html)
+[Radio,](https://docs.flutter.io/flutter/material/Radio-class.html) and
+[Slider](https://docs.flutter.io/flutter/material/Slider-class.html) are
+examples of stateful widgets, which subclass
+[StatefulWidget](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html).
```
Comment (Hixie) [inline tutorials/interactive/index.md]: i think it's misleading to use these as examples of stateful widgets, since they are only stateful because they're animated, not because they are interactive.

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
(Happy to chat whenever.)
